### ISSUE TITLE
Enhancement: option to save TF plan as a workflow artifact

### ISSUE THREAD

**Issue by patheard** [cds-snc/terraform-plan#313]
# Summary
Add an option that causes the action to save the `.tfplan` as a workflow artifact.  This can then be used when the PR merges to main to test that the TF plan is still fresh and prevent unexpected infrastructure changes.

# Related
- https://itnext.io/github-actions-terraform-deployments-with-a-review-of-planned-changes-30143358bb5c

**Comment by patheard** [comment]
Yeah, there is definitely that risk of getting unexpected changes with a stale plan.  Unfortunately  (as you can see by the issue date on this) it has been hanging around for a while.

If you've got time, I'd be more than happy to review a PR that adds the behaviour!

**Comment by eehret** [comment]
Thanks @patheard .   I can't make any promises as I've never authored a Github action before, but I'm certainly interested in helping out if I can.   Will definitely send a PR your way if I manage to get something going! :)
